pandas/io/pytables.py

```diff
@@ -729,10 +729,11 @@ def get(self, key: str):
         object
             Same type as object stored in file.
         """
-        group = self.get_node(key)
-        if group is None:
-            raise KeyError(f"No object named {key} in the file")
-        return self._read_group(group)
+        with patch_pickle():
```

Comment:
seems pretty invasive to do this here. can we not do this in the pickle_compat.py itself?

Reply:
the unpickling is done with a call to `pickle.loads` from within pytables (not _our_ pytables code, but pytables' own code).  the effect of this patching here is to make it use our pickle_compat code instead of the stdlib version.  if there's a less invasive way to do this thatd be great, but i haven't found it

Reply:
ok i understand, can you add some comments here to that effect